models/intermediate/int_ga4_export__derived_event_fields.sql

```diff
@@ -29,7 +25,7 @@ with event_base as (
 
     select 
         *,
-        lag(event_timestamp) over (partition by user_pseudo_id, platform order by event_timestamp) as previous_event_timestamp
+        lag(event_timestamp) over (partition by user_pseudo_id, platform, source_relation order by event_timestamp) as previous_event_timestamp
```

Comment:
Source relation was missed in a few spots, so adding it in throughout the models where necessary.